mmdet3d/core/bbox/coders/monoflex_bbox_coder.py

```diff
@@ -0,0 +1,387 @@
+import numpy as np
+import torch
+from torch.nn import functional as F
+
+from mmdet.core.bbox import BaseBBoxCoder
+from mmdet.core.bbox.builder import BBOX_CODERS
+
+
+@BBOX_CODERS.register_module()
+class MonoFlexCoder(BaseBBoxCoder):
+    """Bbox Coder for MonoFlex.
+
+    Args:
+        depth_mode (str): The depth mode for direct depth calculation.
+        base_depth (tuple[float]): Depth references for decode box depth.
+        depth_range (str): Depth range of predicted depth.
+        use_combined_depth (bool): Whether to use combined depth (direct depth
+            and depth from keypoints).
+        dims_mean (tuple[tuple[float]]): Dimensions mean of decode bbox
+            dimensions [l, h, w] for each category.
+        dims_mean (tuple[tuple[float]]): Dimensions std of decode bbox
+            dimensions [l, h, w] for each category.
```

Comment:
Merge mean and std as a tuple[tuple[float]] like FCOS3DBBoxCoder and name it as base_dims? Just extend the definition of base_depth